admin-dev/themes/default/template/controllers/images/content.tpl

```diff
@@ -123,10 +123,23 @@
 				</div>
 			</div>
 			<div class="panel-footer">
-				<button type="submit" name="submitRegenerate{$table}" class="btn btn-default pull-right" onclick="return confirm('{l s='Are you sure?' d='Admin.Notifications.Warning'}');">
+        <input type="hidden" name="submitRegenerate{$table}" value="" />
+				<button id="submitRegenerate{$table}" class="btn btn-default pull-right">
 					<i class="process-icon-cogs"></i> {l s='Regenerate thumbnails' d='Admin.Design.Feature'}
 				</button>
 			</div>
 		</div>
 	</form>
+
+  <script type="text/javascript">
+    $(function() {
+      $('#submitRegenerate{$table}').on('click', function() {
+        $('#modalRegenerateThumbnails').modal('show');
+        return false;
+      });
+      $('.btn-regenerate-thumbnails').on('click', function () {
+        $('#display_regenerate_form').trigger('submit');
+      });
+    });
+  </script>
```

Comment:
Why didn't you choose to implement it inside a JS file directly? Is there a valid reason?

Reply:
Yes, it's legacy on the first part, and I use a smarty variable.